plugins/webhooks/helper.js

```diff
@@ -0,0 +1,1192 @@
+ */
+async function pushEvent(request, h, parsed, skipMessage, token) {
+    const { eventFactory, pipelineFactory, userFactory } = request.server.app;
+    const { hookId, checkoutUrl, branch, scmContext, type, action, changedFiles, releaseName, ref } = parsed;
+    const fullCheckoutUrl = `${checkoutUrl}#${branch}`;
+    const scmConfig = {
+        scmUri: '',
+        token: '',
+        scmContext
+    };
+
+    request.log(['webhook', hookId], `Push for ${fullCheckoutUrl}`);
+
+    try {
+        scmConfig.token = token;
+        scmConfig.scmUri = await pipelineFactory.scm.parseUrl({
+            checkoutUrl: fullCheckoutUrl,
+            token,
+            scmContext
+        });
+
+        const pipelines = await triggeredPipelines(
+            pipelineFactory,
+            scmConfig,
+            branch,
+            type,
+            action,
+            changedFiles,
+            releaseName,
+            ref
+        );
+        let events = [];
+
+        if (!pipelines || pipelines.length === 0) {
+            request.log(['webhook', hookId], `Skipping since Pipeline ${fullCheckoutUrl} does not exist`);
+        } else {
+            events = await createEvents(
+                eventFactory,
+                userFactory,
+                pipelineFactory,
+                pipelines,
+                parsed,
+                skipMessage,
+                scmConfig
+            );
+        }
+
+        const hasBuildEvents = events.filter(e => e.builds !== null);
+
+        if (hasBuildEvents.length === 0) {
+            return h.response({ message: 'No jobs to start' }).code(204);
+        }
+
+        hasBuildEvents.forEach(e => {
+            request.log(['webhook', hookId, e.id], `Event ${e.id} started`);
+        });
+
+        return h.response().code(201);
+    } catch (err) {
+        logger.error(`[${hookId}]: ${err}`);
+
+        throw err;
+    }
+}
+
+/** Execute scm.getCommitRefSha()
+ * @method getCommitRefSha
+ * @param    {Object}     scm
+ * @param    {String}     token            The token used to authenticate to the SCM
+ * @param    {String}     ref              The reference which we want
+ * @param    {String}     checkoutUrl      Scm checkout URL
+ * @param    {String}     scmContext       Scm which pipeline's repository exists in
+ * @returns  {Promise}                     Specific SHA1 commit to start the build with
+ */
+async function getCommitRefSha({ scm, token, ref, refType, checkoutUrl, scmContext }) {
+    // For example, git@github.com:screwdriver-cd/data-schema.git => screwdriver-cd, data-schema
+    const owner = CHECKOUT_URL_SCHEMA_REGEXP.exec(checkoutUrl)[2];
+    const repo = CHECKOUT_URL_SCHEMA_REGEXP.exec(checkoutUrl)[3];
+
+    return scm.getCommitRefSha({
+        token,
+        owner,
+        repo,
+        ref,
+        refType,
+        scmContext
+    });
+}
+
+/**
+ * Start pipeline events from scm webhook config
```

Comment:
same here but I don't know the same is correct.
```suggestion
 * Start pipeline events with scm webhook config via queue-service
```

Reply:
I fixed it.
Since this function is not necessarily called via queue-service, the phrase `via queue-service` has been omitted.